src/core/mac/mac.cpp

```diff
@@ -944,14 +945,23 @@ bool Mac::IsJoinable(void) const
     return (numUnsecurePorts != 0);
 }
 
-void Mac::ProcessTransmitSecurity(TxFrame &aFrame, bool aProcessAesCcm)
+void Mac::ProcessTransmitSecurity(TxFrame &aFrame)
 {
     KeyManager &      keyManager = Get<KeyManager>();
     uint8_t           keyIdMode;
-    const ExtAddress *extAddress = NULL;
+    const ExtAddress *extAddress            = NULL;
+    bool              processTransmitAesCcm = true;
 
     VerifyOrExit(aFrame.GetSecurityEnabled(), OT_NOOP);
 
+#if OPENTHREAD_CONFIG_TIME_SYNC_ENABLE
+    if (aFrame.GetTimeIeOffset() != 0)
+    {
+        // Transmit security will be processed after time IE content is updated.
+        processTransmitAesCcm = false;
+    }
+#endif
```

Comment:
```suggestion
#if OPENTHREAD_CONFIG_TIME_SYNC_ENABLE
VerifyOrExit(aFrame.GetTimeIeOffset() == 0);
#endif
```
And move to just before: `aFrame.ProcessTransmitAesCcm(*extAddress);`